tree/homework2.php

```diff
@@ -0,0 +1,75 @@
+<?php
+require_once ('tree_class.php');
+
+class Queue
+{
+    private $elements;
+
+    public function __construct()
+    {
+        $this->elements = array();
+    }
+
+    public function enqueue($num) {
+        array_unshift($this->elements, $num);
+    }
+
+    public function dequeue() {
+        if (!$this->isEmpty()) {
+            array_pop($this->elements);
```

Comment:
Because method `array_pop()`  return the element off the end of the array, so you should return it right here.  Then method `dequeue` can use as a combination of `front() + dequeue()`, and you can remove `front()`. It's called optimize your class.

```php
class Queue
{
    private $elements;

    public function __construct() {/* ... */}

    public function enqueue($num) {/* ... */}

    public function dequeue() {
        if (!$this->isEmpty()) {
            return array_pop($this->elements);
        }
        return null;
    }

    public function isEmpty() {/* ... */}
}
```

Reply:
I'll re-implement `Queue` class according to this.